src/Hash.cc

```diff
@@ -206,16 +197,273 @@ void* HashKey::TakeKey()
 		return CopyKey(key, size);
 	}
 
-void* HashKey::CopyKey(const void* k, int s) const
+char* HashKey::CopyKey(const char* k, size_t s) const
 	{
-	void* k_copy = (void*) new char[s];
+	char* k_copy = new char[s];
 	memcpy(k_copy, k, s);
 	return k_copy;
 	}
 
-hash_t HashKey::HashBytes(const void* bytes, int size)
+hash_t HashKey::HashBytes(const void* bytes, size_t size)
 	{
 	return KeyedHash::Hash64(bytes, size);
 	}
 
+void HashKey::Set(bool b)
+	{
+	Set(b);
```

Comment:
Should this be `Set(static_cast<int>(b))`?

Reply:
Thanks for spotting this — I noticed yesterday because I ended up calling this and ... yeah. It's an edit bug from a version of the code I had earlier that indeed was casting. Since I ended up added `bool` support all the way through into the `key_u` union, I was thinking of making it similar to the other ones, i.e.:

```
void HashKey::Set(bool b)
        {
        key_u.b = b;
        key = reinterpret_cast<char*>(&key_u);
        size = sizeof(b);
        }
```